django/db/migrations/executor.py

```diff
@@ -179,19 +179,24 @@ def unapply_migration(self, state, migration, fake=False):
     def detect_soft_applied(self, project_state, migration):
         """
         Tests whether a migration has been implicitly applied - that the
-        tables it would create exist. This is intended only for use
-        on initial migrations (as it only looks for CreateModel).
+        tables or columns it would create exist. This is intended only for use
+        on initial migrations (as it only looks for CreateModel and AddField).
         """
-        # Bail if the migration isn't the first one in its app
-        if [name for app, name in migration.dependencies if app == migration.app_label]:
+
+        # Bail if it's NOT an initial migration
+        if migration.initial is None:
+            if any(app == migration.app_label for app, name in migration.dependencies):
+                return False, project_state
+        elif migration.initial is False:
             return False, project_state
+
         if project_state is None:
             after_state = self.loader.project_state((migration.app_label, migration.name), at_end=True)
         else:
             after_state = migration.mutate_state(project_state)
         apps = after_state.apps
         found_create_migration = False
-        # Make sure all create model are done
+        # Make sure all create model and add field operations are done
```

Comment:
Could you replace "create model" and "add field" with "CreateModel" and "AddField" respectively, please.